apps/internal/oauth/resolvers.go

```diff
@@ -47,7 +47,9 @@ func newAuthorityEndpoint(rest *ops.REST) *authorityEndpoint {
 // ResolveEndpoints gets the authorization and token endpoints and creates an AuthorityEndpoints instance
 func (m *authorityEndpoint) ResolveEndpoints(ctx context.Context, authorityInfo authority.Info, userPrincipalName string) (authority.Endpoints, error) {
 	if authorityInfo.AuthorityType == ADFS && len(userPrincipalName) == 0 {
-		return authority.Endpoints{}, errors.New("UPN required for authority validation for ADFS")
+		if authorityInfo.ValidateAuthority {
```

Comment:
Authority validation and instance discovery are different things, but on AAD we use the same mechanism to implement both. 

Authority validation - make a call to instance discovery endpoint to check that the authority host is valid. Only makes sense for AAD. For ADFS the mechanism was deemed unworkable. For B2C it doesn't exist.

Instance discovery - is also used to find authority aliases. E.g. "login.microsoft.com" is the same as "login.windows.net". This helps MSAL with caching. 


@rayluo can advise here, but the recommendation is to have a flag `instanceDIscoveryEnabled` that disables both concepts. In the future, we should deprecate / remove the concept of authority validation.

Reply:
removing the code to validate authority for ADFS